src/App.js

```diff
@@ -17,8 +18,25 @@ const tabs = [
   },
 ];
 
-const App = () => (
-  <h1>React tabs</h1>
-);
+const preparedTabs = sourceTabs.map((tab, index) => ({
+  ...tab,
+  index,
+}));
+
+class App extends React.Component {
+  state = {
+    tabs: preparedTabs,
```

Comment:
you can use just preparedTabs, no need to put them in the state, cause you don't modify them in the component